File: Logic Building Problems/Easy Problems/Check if a number is a power of another number/method2.cpp
// CPP program to check given number y
// is power of x
#include <iostream>
#include <math.h>
using namespace std;

bool isPower(int x, int y)
{
	// logarithm function to calculate value
	float res1 = log(y) / log(x);
	return res1==floor(res1);
}

// Driven program
int main()
{
	cout << isPower(2, 128) << endl;
	return 0;
}
 

File: Logic Building Problems/Easy Problems/Check if a number is a power of another number/method2.js
// JavaScript program to check given number y
// is power of x

function isPower(x, y) {
	// logarithm function to calculate value
	const res1 = Math.log(y) / Math.log(x);
	const res2 = Math.log(y) / Math.log(x);
	
	// compare to the result1 or result2 both are equal
	return (res1 === res2);
}

// Driven program
console.log(isPower(2, 128));
